generator/src/main/java/org/corfudb/generator/State.java

```diff
@@ -0,0 +1,110 @@
+package org.corfudb.generator;
+
+import com.google.common.reflect.TypeToken;
+
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.UUID;
+
+import org.corfudb.generator.distributions.Keys;
+import org.corfudb.generator.distributions.OperationCount;
+import org.corfudb.generator.distributions.Operations;
+import org.corfudb.generator.distributions.Streams;
+import org.corfudb.runtime.CorfuRuntime;
+import org.corfudb.runtime.collections.SMRMap;
+import org.corfudb.runtime.object.transactions.TransactionType;
+
+import lombok.Getter;
+import lombok.Setter;
+
+/**
+ * This object keeps state information of the different data distributions and runtime client.
+ *
+ * Created by maithem on 7/14/17.
+ */
+public class State {
+
+    @Getter
+    final Streams streams;
+
+    @Getter
+    final Keys keys;
+
+    @Getter
+    final OperationCount operationCount;
+
+    @Getter
+    final Operations operations;
+
+    @Getter
+    final CorfuRuntime runtime;
+
+    @Getter
+    final Map<UUID, SMRMap> maps;
+
+    @Getter
+    @Setter
+    volatile long trimMark = -1;
+
+    public State(int numStreams, int numKeys, CorfuRuntime rt) {
+        streams = new Streams(numStreams);
+        keys = new Keys(numKeys);
+        operationCount = new OperationCount();
+
+        this.runtime = rt;
+        maps = new HashMap<>();
+        operations = new Operations(this);
+
+        streams.populate();
+        keys.populate();
+        operationCount.populate();
+        operations.populate();
+
+        openObjects();
+    }
+
+    private void openObjects() {
+        for (UUID uuid : streams.getDataSet()) {
+            SMRMap<UUID, UUID> map = runtime.getObjectsView()
+                    .build()
+                    .setStreamID(uuid)
+                    .setTypeToken(new TypeToken<SMRMap<UUID,UUID>>() {})
+                    .open();
+
+            maps.put(uuid, map);
+        }
+    }
+
+    public Map<String, String> getMap(UUID uuid) {
+        Map map = maps.get(uuid);
+        if (map == null) {
+            throw new RuntimeException("Map doesn't exist");
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Define and throw a dedicated exception instead of using a generic one. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00112)